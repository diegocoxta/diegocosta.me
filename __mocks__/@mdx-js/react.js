module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
