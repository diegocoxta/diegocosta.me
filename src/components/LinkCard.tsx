import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Icon from '~/components/Icon';
import { IoMdOpen } from 'react-icons/io';

const Container = styled.a.attrs({
  target: '_blank',
})`
  background: ${({ theme }) => theme.accentColor};
  border-radius: 30px;
  margin: 16px 0;
  display: block;
  text-decoration: none;
  cursor: pointer;
  transition: all 0.5s;

  &:hover {
    transform: scale(1.05);
  }
`;

const Button = styled.div`
  display: flex;
  padding: 10px 16px;
  color: #fff;
  font-weight: bold;
  justify-content: space-between;
  align-items: center;
  line-height: 1;
`;

const ButtonIcon = styled.div`
  margin: 2px 5px 0 0;
`;

const ButtonContent = styled.div`
  display: flex;
  align-items: center;
`;

const ButtonDescription = styled.span`
  font-size: 11px;
  font-weight: 300;
  margin: 0 10px;
`;

const CardContent = styled.div`
  background: ${({ theme }) => theme.titleColor};
  padding: 10px 16px 16px;
  font-size: 16px;
  line-height: 1.4;
  color: ${({ theme }) => theme.backgroundColor};
  border-radius: 0 0 30px 30px;

  p {
    margin: 0;
    display: inline;
  }

  span {
    font-size: 14px;
    display: block;
    margin-top: 10px;
  }
`;

interface LinkCardProps {
  url: string;
  title: string;
  api?: string | null;
  icon?: string | null;
}

export default function LinkCard(props: LinkCardProps) {
  const [content, setContent] = useState('');

  useEffect(() => {
    if (props.api) {
      fetch(props.api).then(async (response) => {
        const json = await response.json();
        setContent(json.message);
      });
    }
  }, []);

  return (
    <Container href={props.url}>
      <Button>
        <ButtonContent>
          {props.icon && (
            <ButtonIcon>
              <Icon name={props.icon} />
            </ButtonIcon>
          )}{' '}
          {props.title}
          <ButtonDescription>
            {props.url.replace('https://', '').replace('www.', '').replace('mailto:', '')}
          </ButtonDescription>
        </ButtonContent>
        <IoMdOpen />
      </Button>
      {props.api && content && (
        <CardContent data-testid="linkcard-card-content" dangerouslySetInnerHTML={{ __html: content }}></CardContent>
      )}
    </Container>
  );
}
