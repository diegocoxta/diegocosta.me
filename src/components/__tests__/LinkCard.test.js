import React from 'react';
import { render } from '@testing-library/react';

import LinkCard from '../LinkCard';

describe('<LinkCard />', () => {
  beforeEach(() => {
    global.fetch = jest.fn().mockImplementation();
    jest.spyOn(global, 'fetch').mockResolvedValue({
      json: jest.fn().mockResolvedValue({ message: 'Hello World' }),
    });
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it('renders properly', async () => {
    const { baseElement, findByText } = render(
      <LinkCard url="https://google.com" title="Google Inc." api="/api/my-api" />
    );

    expect(await findByText('Google Inc.')).toBeTruthy();
    expect(await findByText('google.com')).toBeTruthy();
    expect(await findByText('Hello World')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
