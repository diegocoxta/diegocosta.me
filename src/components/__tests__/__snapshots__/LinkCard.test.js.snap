// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkCard /> renders properly 1`] = `
.c0 {
  border-radius: 30px;
  margin: 16px 0;
  display: block;
  text-decoration: none;
  cursor: pointer;
  transition: all 0.5s;
}

.c0:hover {
  transform: scale(1.05);
}

.c1 {
  display: flex;
  padding: 10px 16px;
  color: #fff;
  font-weight: bold;
  justify-content: space-between;
  align-items: center;
  line-height: 1;
}

.c2 {
  display: flex;
  align-items: center;
}

.c3 {
  font-size: 11px;
  font-weight: 300;
  margin: 0 10px;
}

.c4 {
  padding: 10px 16px 16px;
  font-size: 16px;
  line-height: 1.4;
  border-radius: 0 0 30px 30px;
}

.c4 p {
  margin: 0;
  display: inline;
}

.c4 span {
  font-size: 14px;
  display: block;
  margin-top: 10px;
}

<body>
  <div>
    <a
      class="c0"
      href="https://google.com"
      target="_blank"
    >
      <div
        class="c1"
      >
        <div
          class="c2"
        >
           
          Google Inc.
          <span
            class="c3"
          >
            google.com
          </span>
        </div>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405.34 405.332H106.66V106.668H240V64H106.66C83.191 64 64 83.197 64 106.668v298.664C64 428.803 83.191 448 106.66 448h298.68c23.469 0 42.66-19.197 42.66-42.668V272h-42.66v133.332zM288 64v42.668h87.474L159.999 322.133l29.866 29.866 215.476-215.47V224H448V64H288z"
          />
        </svg>
      </div>
      <div
        class="c4"
        data-testid="linkcard-card-content"
      >
        Hello World
      </div>
    </a>
  </div>
</body>
`;
